config/cmake/MFEMConfig.cmake.in

```diff
@@ -63,6 +63,7 @@ set(MFEM_USE_ALGOIM @MFEM_USE_ALGOIM@)
 set(MFEM_USE_BENCHMARK @MFEM_USE_BENCHMARK@)
 set(MFEM_USE_PARELAG @MFEM_USE_PARELAG@)
 set(MFEM_USE_ENZYME @MFEM_USE_ENZYME@)
+set(MFEM_USE_FLOAT @MFEM_USE_FLOAT@)
```

Comment:
This really means "MFEM_USE_SINGLE_PRECISION" right?

Reply:
We probably need to adjust this, but basically the idea in this branch is that we can replace all doubles with another type e.g. float

Reply:
Yes, it means to use `float` as the floating point type, see `general/globals.hpp`. We may also want to add options for `long double` and maybe even half precision for the most adventurous users. The changes are mostly general, so that any floating point type could be used, but a few places requires specially defined constants (e.g. tolerances for Newton solvers), so we may need different flags for different precision levels. I'm sure there will be opinions about names for these flags.

Reply:
That means the `MFEM_USE_XXX` macros are not the right way to provide this option. It's more convenient to provide a build macro like `MFEM_FP_TYPE=XXX` where the person who builds MFEM puts the type. This would save quite a bit of ifdefs in `general/globals.hpp`.